numpy/lib/financial.py

```diff
@@ -573,15 +570,26 @@ def rate(nper, pmt, pv, fv, when='end', guess=0.10, tol=1e-6, maxiter=100):
 
     """
     when = _convert_when(when)
+    default_type = Decimal if isinstance(pmt, Decimal) else float
+
+    # Handle casting defaults to Decimal if/when pmt is a Decimal and
+    # guess and/or tol are not given default values
+    if guess is None:
+        guess = default_type('0.1')
+
+    if tol is None:
+        tol = default_type('1e-6')
+
     (nper, pmt, pv, fv, when) = map(np.asarray, [nper, pmt, pv, fv, when])
+
     rn = guess
-    iter = 0
```

Comment:
`iter` is a keyword, so I changed it to iterator.

Reply:
Not a keyword, just a builtin. Still sensible though.